### PR TITLE
Sacrificial nao

### LINKED ISSUE #984: Partial Visual Referee with sacrificial NAO
Only run pose detection on the NAO/NAOs with the best view of the referee. If one detects the visual referee signal, only that NAO goes to Ready. It then checks, if it is penalized for a certain period of time. After the time, if it hasn't been penalized, it sends a message to all other NAO, telling them to go to Ready.

### PR DESCRIPTION
**Pull request by Vivituhh**
## Why? What?

Re-enables message sending in `Standby` after referee detection, to enable transition to `Ready` also for robots which do not see referee.
Moreover it implements an sacrificial lamb for the referee pose detection, to avoid penalizing of whole team after false detection.

Fixes #984

## ToDo / Known Issues

*If this is a WIP describe which problems are to be fixed.*

## Ideas for Next Iterations (Not This PR)

*If there are some improvements that could be done in a next iteration, describe them here.*

## How to Test

Test at least with 2 robots (1 detecting, 1 sacrificial lamb at the moment playernumber 3). 
Both should wait 5s after detection (wait for penalty of opponent team), then the lamb transitions to `Ready` and if it is not penalized the others transition to `Ready` 20s after detection.

### CODE REVIEW

File: crates/control/src/referee_pose_detection_filter.rs [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+    pose_detection::VisualRefereeState,
     pose_kinds::PoseKind,
 };
 
 #[derive(Deserialize, Serialize)]
 pub struct RefereePoseDetectionFilter {
     detection_times: Players<Option<SystemTime>>,
     detected_above_arm_poses_queue: VecDeque<bool>,
+    visual_referee_state: VisualRefereeState,
+    motion_in_standby_count: usize,
```
Comment (oleflb) [inline crates/control/src/referee_pose_detection_filter.rs:28]: switch rows

**Review by MaikRe — CHANGES_REQUESTED** [review]

File: crates/control/src/sacrificial_lamb.rs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                            matches!(player.penalty, Some(Penalty::IllegalMotionInStandby { .. }))
+                        })
+                        .count()
+            })
+            .max();
+
+        let motion_in_standby =
+            new_motion_in_standby_count.map_or(false, |new_motion_in_standby_count| {
+                let motion_in_standby = new_motion_in_standby_count > self.motion_in_standby_count;
+                self.motion_in_standby_count = new_motion_in_standby_count;
```
[the remark this answers is not in the document]
Comment (MaikRe) [reply crates/control/src/sacrificial_lamb.rs:89]: Nevermind on this, as decrease also accounts for the game controller controller making mistakes which are probably more likely than messages being sent/received in the wrong order

File: crates/control/src/referee_pose_detection_filter.rs (hunk 1 above)
Comment (Vivituhh) [reply crates/control/src/referee_pose_detection_filter.rs:28]: Done

File: crates/control/src/referee_pose_detection_filter.rs [hunk 3]
```diff
@@ -184,3 +220,17 @@ fn unpack_own_detection_tree(
         })
         .collect()
 }
+
+fn send_own_detection_message<T: NetworkInterface>(
+    hardware_interface: Arc<T>,
+    player_number: PlayerNumber,
+    time_to_reach_kick_position: Duration,
+) -> Result<()> {
+    hardware_interface.write_to_network(dbg!(OutgoingMessage::Spl(HulkMessage {
```
Comment (MaikRe) [inline crates/control/src/referee_pose_detection_filter.rs:229]: Please no debug

Comment (Vivituhh) [reply crates/control/src/referee_pose_detection_filter.rs]: Done

**Review by MaikRe — APPROVED** [review]
Noice

**Pull request merged**